src/app/@ansyn/ol/plugins/annotations/feature-properties-validation.ts

```diff
@@ -0,0 +1,45 @@
+import { cloneDeep } from "lodash";
+import { Feature } from "geojson";
+import { MarkerSize } from "@ansyn/imagery";
+
+export function validateFeatureProperties(feature: Feature<any>): Feature<any> {
+	const featureJson = cloneDeep(feature);
+	const { id, style, label, labelSize, icon, showMeasures, showArea, undeletable, mode, labelTranslateOn } = featureJson.properties;
+	const { opacity, initial } = !!style ? style :  { opacity: null, initial: null };
+
+	featureJson.properties = {
+		id: (!!id && typeof id === 'string') ? id : '',
+		labelSize: (!!labelSize && typeof labelSize === 'number') ? labelSize : 28,
```

Comment:
why don't you use the getInitialAnnotationsFeatureProperties?

Reply:
i think the validate function should be written in @ansyn/imagery/model/visualizers/initial-annotations-feature-properties.ts  file (so it can be used by cesium as well)